### DISCUSSION TITLE
How to implement Stochastic Weight Averaging?

### DISCUSSION THREAD

**Discussion by nalzok** [google-deepmind/optax#389] [Q&A]
I want to implement [Stochastic Weight Averaging](https://pytorch.org/blog/pytorch-1.6-now-includes-stochastic-weight-averaging#how-does-stochastic-weight-averaging-work) with Optax. The idea is simple: Use something like Adam for some epochs, and switch to a constant learning rate after say 100 steps. See the following figure for an illustration:

![](https://pytorch.org/assets/images/nswapytorch2.jpg)
